### PR TITLE
ft(bus-routes):implementing bus CRUD operations

### PR DESCRIPTION
**Pull request by Isaackomeza**
-user can create, read, update or delete buses

#### What does this PR do?

- Creating a CRUD operation for buses

#### Description of Task to be completed?

- Creating a database 
- Using sequelize to generate models, migrations, and seeders
- Creating all bus routes
- Testing bus endpoints with jest

#### How should this be manually tested?

- Use postman to test bus endpoints

#### Any background context you want to provide?

User should be able to 

- CREATE Bus
- GET Buses
- GET Bus
- UPDATE Bus
- DELETE Bus

#### What are the relevant trello boards stories?

- https://trello.com/c/y8gAasfO/109-crud-operations-for-bus

#### Screenshots (if appropriate)

- N/A

#### Questions:

- N/Q

### CODE REVIEW

**Comment by Isaackomeza** [comment]
[CRUD Operations for Bus](https://trello.com/c/y8gAasfO/109-crud-operations-for-bus)

**Review by Rwandarushya — APPROVED** [review]

**Review by Imaraika — APPROVED** [review]

File: src/controllers/busControllers.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  try {
+    const { busId } = req.params;
+    const [updated] = await models.Bus.update(req.body, {
+      where: { id: busId },
+    });
+    if (updated) {
+      const updatedBus = await models.Bus.findOne({ where: { id: busId } });
+      return res.status(200).json({ status: 200, message: 'Bus was updated successfully.', bus: updatedBus });
+    }
+    return res.status(404).json({ status: 404, message: `Cannot update Bus with id ${busId}. Bus was not found!` });
```
Comment (malaba6) [inline src/controllers/busControllers.js]: It would be enough to just say: Bus with ID (id) was not found

File: src/helper/bus.inputs.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      const long = coordinates[1].trim();
+      const valRegex = /-?\d/;
+      if (!valRegex.test(lat)) {
+        return helper.message('Laltitude must be numbers');
+      }
+      if (!valRegex.test(long)) {
+        return helper.message('Longitude must be numbers');
+      }
+    }),
+    bus_status: Joi.string().required().valid('active', 'inactive'),
```
Comment (malaba6) [inline src/helper/bus.inputs.js]: Actually, bus_status should not be required. If not provided, it should default to inactive

File: src/controllers/busControllers.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
+import models from '../database/models';
+
+const createBus = async (req, res) => {
+  try {
+    const bus = await models.Bus.create(req.body);
+    return res.status(201).json({ status: 201, message: 'Bus created successfully', bus });
+  } catch (error) {
+    return res.status(500).json({ error: error.message });
+  }
```
Comment (malaba6) [inline src/controllers/busControllers.js:13]: I don't see where you handle duplication here. When the user tries to create the same resource twice

**Review by malaba6 — CHANGES_REQUESTED** [review]

File: src/controllers/busControllers.js (hunk 1 above)
Comment (Isaackomeza) [reply src/controllers/busControllers.js]: The issue is now fixed

File: src/helper/bus.inputs.js (hunk 2 above)
Comment (Isaackomeza) [reply src/helper/bus.inputs.js]: Yes, done

File: src/controllers/busControllers.js (hunk 3 above)
Comment (Isaackomeza) [reply src/controllers/busControllers.js:13]: It is handled now, and Thank you for reminding me that

**Pull request merged**